src/Pages/RegisterForm/RegisterForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react';
+import { Button, Form, Input } from 'antd';
+import type { FormItemProps } from 'antd';
+import './style.scss';
+
+const MyFormItemContext = React.createContext<(string | number)[]>([]);
+
+interface MyFormItemGroupProps {
+  prefix: string | number | (string | number)[];
+}
+
+function toArr(
+  str: string | number | (string | number)[]
+): (string | number)[] {
+  return Array.isArray(str) ? str : [str];
+}
+
+const MyFormItemGroup: React.FC<
+  React.PropsWithChildren<MyFormItemGroupProps>
+> = ({ prefix, children }) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatPath = React.useMemo(
+    () => [...prefixPath, ...toArr(prefix)],
+    [prefixPath, prefix]
+  );
+
+  return (
+    <MyFormItemContext.Provider value={concatPath}>
+      {children}
+    </MyFormItemContext.Provider>
+  );
+};
+
+const MyFormItem = ({ name, ...props }: FormItemProps) => {
+  const prefixPath = React.useContext(MyFormItemContext);
+  const concatName =
+    name !== undefined ? [...prefixPath, ...toArr(name)] : undefined;
+
+  return <Form.Item name={concatName} {...props} />;
+};
+
+const App: React.FC = () => {
+  const [form] = Form.useForm();
+
+  const [formData, setFormData] = useState({
+    firstName: '',
+    businessEmail: '',
+    companyName: '',
+  });
+  const [errorName, setErrorName] = useState<string | null>(null);
+  const [errorEmail, setErrorEmail] = useState<string | null>(null);
+  const [errorCompany, setErrorCompany] = useState<string | null>(null);
+
+  const validateForm = () => {
+    let valid = true;
+
+ 
+    if (!/^[\p{L}\d\s._-]+$/u.test(formData.firstName)) {
+      setErrorName('Please enter only letters and the field cannot be empty');
+      valid = false;
+    } else {
+      setErrorName(null);
+    }
+
+
+    if (
+      !/^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/.test(
+        formData.businessEmail
+      )
+    ) {
+      setErrorEmail('Enter correct e-mail');
+      valid = false;
+    } else {
+      setErrorEmail(null);
+    }
+
+ 
+    if (!/^[\p{L}\d\s._-]+$/u.test(formData.companyName)) {
+      setErrorCompany('Enter correct company name');
+      valid = false;
+    } else {
+      setErrorCompany(null);
+    }
+
+    return valid;
+  };
+
+  const onFinish = () => {
+    if (validateForm()) {
+      console.log('Form submitted:', formData);
+    
+
+      form.resetFields();
+    } else {
+      console.log('Form contains errors.');
+    }
+  };
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+    field: string
+  ) => {
+    const { value } = event.target;
+    setFormData(prevData => ({
+      ...prevData,
+      [field]: value,
+    }));
+  };
+
+  return (
+    <div className="form">
```

Comment:
I would double check styling on the form inouts. (I tried to enter and i did not see a letter . Unless i selected my input. Just check it out. 

selected  input 
<img width="168" alt="image" src="https://github.com/user-attachments/assets/5dc7ca9f-fc2f-4073-ad39-8a23c3bb04e3">


with inout but UNSELECTED:
<img width="232" alt="image" src="https://github.com/user-attachments/assets/9d48e4f6-d09f-4217-b183-e85615259ca6">

Reply:
Changed styles